### ISSUE TITLE
Reset() form that has validation errors

### ISSUE THREAD

**Issue by Khorkhe** [erikras/redux-form#2714]
In a form with validation functions specified, I have a *reset form* button that calls `this.props.reset()` to clear all form fields:

```
<Button id='clear' type='reset' icon='remove' onClick={this.closeAndReset} />
```

where the onClick function is:
```
closeAndReset = () => {
    console.log("inside closeAndReset")
    this.props.reset()
    // other things
  }
```

Assuming the below validation function:

```
const asyncValidate = (values) => {
  return axios.get(something(values.label).url)
  .then((response) => {
    if (condition_fails)
      throw { label: 'Async error message' }
  })
}
```

If a user enters some incorrect value in the *label* field, and then presses the button that triggers the *form reset*, the form will not reset.  The console log statement is not executed either, so it seems the onClick event is not triggered. Instead, it will throw the validation error above.  The second *reset button* click will then reset the form.

Basically, the *reset button* click causes an onBlur event of the *label* field, which triggers the validation prior to the *reset*.   Is there a commonly accepted solution for this?

**Comment by Khorkhe** [comment]
the problem turned out unrelated to Redux Form. Here is actually what happens:

 - click on field that has "Required" validation
 - click on the "reset" button, **located at the bottom of the form**
 - validation is triggered
 - validation causes a "Required" label to appear below the concerned field
 - **onClick should be triggered here**, but the error label that was created by the validation caused the elements below it to be pushed down.  This causes the "reset" button to be pushed below the mouse cursor, and the click event happens outside of the button.
